lib/Dialect/Substrait/IR/Substrait.cpp

```diff
@@ -116,6 +116,28 @@ LogicalResult mlir::substrait::IntervalDaySecondAttr::verify(
   return success();
 }
 
+LogicalResult mlir::substrait::ListAttr::verify(
+    llvm::function_ref<mlir::InFlightDiagnostic()> emitError, ArrayAttr value,
+    ListType listType) {
+
+  mlir::Type expectedType =
+      listType.getType(); // The atomic type the list is parameterized with
+
+  for (mlir::Attribute attr : value) {
+    auto typedAttr = mlir::dyn_cast<mlir::TypedAttr>(attr);
+    if (!typedAttr) {
+      return emitError()
+             << "ListAttr values must be typed attributes, but got: " << attr;
```

Comment:
Nit: single-quoted class name.
```suggestion
             << "'ListAttr' values must be typed attributes, but got: " << attr;
```